### PR TITLE
feat(admin-ui): implement audit log on Services plugin

### PR DESCRIPTION
**Pull request by cooldev-blue**

### CODE REVIEW

**Comment by duttarnab** [comment]
1. Ldap save not working after saving comments. Save/Update/Deleted  should work after entering comments
2. Audit logs not updated after Fetch/Save/Update/Deleted

**Comment by cooldev-blue** [comment]
Corrected some features

**Comment by cooldev-blue** [comment]
Merged branch 'main' into issue#134

File: plugins/services-plugin/Components/Configuration/LdapAddPage.js [hunk 1]
```diff
@@ -32,6 +42,7 @@ function LdapAddPage({ dispatch }) {
 }
 const mapStateToProps = (state) => {
   return {
+    scripts: state.ldapReducer.items,
```
Comment (duttarnab) [inline plugins/services-plugin/Components/Configuration/LdapAddPage.js:45]: this is a Ldap config Add page, what is the purpose of getting items from store here?

File: plugins/services-plugin/Components/Configuration/LdapForm.js [hunk 2]
```diff
@@ -13,14 +13,27 @@ import GluuTypeAhead from '../../../../app/routes/Apps/Gluu/GluuTypeAhead'
...
 import GluuLabel from '../../../../app/routes/Apps/Gluu/GluuLabel'
 
-function LdapForm({ item, handleSubmit }) {
+import GluuCommitFooter from '../../../../app/routes/Apps/Gluu/GluuCommitFooter'
+import GluuCommitDialog from '../../../../app/routes/Apps/Gluu/GluuCommitDialog'
+
+function LdapForm({ item, scripts, handleSubmit }) {
     const [init, setInit] = useState(false)
+    const [modal, setModal] = useState(false)
+    // const [scriptTypeState, setScriptTypeState] = useState(item.scriptType)
```
Comment (duttarnab) [inline plugins/services-plugin/Components/Configuration/LdapForm.js:22]: please remove unnecessary comments from the code. I know there are already some redundant comments in code.. We should remove those.

File: plugins/services-plugin/redux/api/LdapApi.js [hunk 3]
```diff
@@ -33,6 +33,7 @@ export default class LdapApi {
   addLdapConfig = (input) => {
     return new Promise((resolve, reject) => {
       this.api.postConfigDatabaseLdap(input, (error, data) => {
+        console.log(data)
```
Comment (duttarnab) [inline plugins/services-plugin/redux/api/LdapApi.js:36]: please remove unnecessary logs from code before check-in

File: plugins/services-plugin/redux/reducers/LdapReducer.js [hunk 4]
```diff
@@ -135,3 +135,4 @@ export default function ldapReducer(state = INIT_STATE, action) {
       }
   }
 }
+reducerRegistry.register(reducerName, ldapReducer)
```
Comment (duttarnab) [inline plugins/services-plugin/redux/reducers/LdapReducer.js:138]: what is the reason for registering the reducer here? Please check how it is done for other reducers.

File: plugins/services-plugin/redux/sagas/LdapSaga.js [hunk 5]
```diff
@@ -20,19 +31,45 @@ import { getClient } from '../../../../app/redux/api/base'
 const JansConfigApi = require('jans_config_api')
 
 function* newFunction() {
-  const token = yield select((state) => state.authReducer.token.access_token)
+  //const token = yield select((state) => state.authReducer.token.access_token)
+  
+  const wholeToken = yield select((state) => state.authReducer.token)
+  console.log('==============Token ' + JSON.stringify(wholeToken))
```
Comment (duttarnab) [inline plugins/services-plugin/redux/sagas/LdapSaga.js:37]: remove unnecessary logs before check-in

File: plugins/services-plugin/redux/actions/LdapActions.js [hunk 6]
```diff
@@ -12,38 +12,39 @@ import {
...
+  payload: { action },
 })
 
 export const getLdapResponse = (data) => ({
   type: GET_LDAP_RESPONSE,
   payload: { data },
 })
 
-export const addLdap = (data) => ({
+export const addLdap = (action) => ({
```
Comment (duttarnab) [inline plugins/services-plugin/redux/actions/LdapActions.js:25]: what is the reason of renaming `data` to `action`

**Review by duttarnab — CHANGES_REQUESTED** [review]

File: app/routes/Apps/Gluu/GluuFooter.js [hunk 7]
```diff
@@ -15,10 +19,16 @@ function GluuFooter({ extraOnClick, extraLabel, hideButtons }) {
               {extraLabel}
             </Button>
           )}
+          <Button color="primary" type="submit" className="LdapUserActionSubmitButton" style={{ visibility: 'hidden' }}>
```
[the remark this answers is not in the document]
Comment (cooldev-blue) [reply app/routes/Apps/Gluu/GluuFooter.js]: I fixed it

File: plugins/services-plugin/Components/Configuration/LdapAddPage.js (hunk 1 above)
Comment (cooldev-blue) [reply plugins/services-plugin/Components/Configuration/LdapAddPage.js]: No need
I removed

File: plugins/services-plugin/Components/Configuration/LdapForm.js (hunk 2 above)
Comment (cooldev-blue) [reply plugins/services-plugin/Components/Configuration/LdapForm.js]: I removed unnecessary comments

File: plugins/services-plugin/redux/api/LdapApi.js (hunk 3 above)
Comment (cooldev-blue) [reply plugins/services-plugin/redux/api/LdapApi.js]: I removed logs

File: plugins/services-plugin/redux/reducers/LdapReducer.js (hunk 4 above)
Comment (cooldev-blue) [reply plugins/services-plugin/redux/reducers/LdapReducer.js]: I fixed it

File: plugins/services-plugin/redux/sagas/LdapSaga.js (hunk 5 above)
Comment (cooldev-blue) [reply plugins/services-plugin/redux/sagas/LdapSaga.js]: I removed logs

File: plugins/services-plugin/redux/actions/LdapActions.js (hunk 6 above)
Comment (cooldev-blue) [reply plugins/services-plugin/redux/actions/LdapActions.js]: It was just renaming
I fixed it

**Review by duttarnab — CHANGES_REQUESTED** [review]
Ldap featch not working for branch issue#134

**Comment by cooldev-blue** [comment]
now ldap page is populating records

**Review by duttarnab — CHANGES_REQUESTED** [review]
Please following error on console.
image

**Comment by cooldev-blue** [comment]
Fixed error by changing payload parameter

**Review by duttarnab — APPROVED** [review]

**Pull request merged**